Lib/python/pyrun.swg

```diff
@@ -582,6 +603,7 @@ SwigPyObject_own(PyObject *v, PyObject *args)
       } else {
         SwigPyObject_disown(v,args);
       }
+      Py_DECREF(Py_None);
```

Comment:
This had me puzzled until I started looking at all the return values of the above functions. For code clarity, I suggest instead to use
```Py_DECREF(SwigPyObject_acquire(v,args));```
and
```Py_DECREF(SwigPyObject_disown(v,args));```

I started looking around for other problems for the functions that use SWIG_Py_Void and the call to SwigPyObject_append in SWIG_Python_InitShadowInstance is another leak you could fix in this patch in a similar way.